### DISCUSSION TITLE
Global Automerge

### DISCUSSION THREAD

**Discussion by SamuelCabralCruz** [renovatebot/renovate#12813] [Request Help]
Hi,

I am trying to set the default value of automerge to true instead of having to override it in each package rules.
However, this configuration does not seem to be taken into account.
Am I doing something wrong?

Thanks in advance,

**Comment by rarkins** [comment]
Probably

**Comment by SamuelCabralCruz** [comment]
@rarkins here is a subset of my renovate configuration file. Is this config supposed to be ok?

```
{
  "extends": ["config:base"],
  "masterIssue": true,
  "schedule": ["before 6am on Wednesday"],
  "automerge": true,
  "regexManagers": [
    {
      "fileMatch": ["(^|/)Makefile$", "^.*\.mk$"],
      "matchStrings": ["@go install (?<depName>[^@]*)@(?<currentValue>.*)"],
      datasourceTemplate: "go",
    },
  ],
  "postUpdateOptions": ["gomodTidy", "gomodUpdateImportPaths"],
  "packageRules": [
    {
      "groupName": "Dependencies Maintenance",
      "matchUpdateTypes": ["pin", "digest"],
      "matchPackagePatterns": [".*"],
      // "automerge": true - should not have to specify
    },
    {
      "groupName": "Dependencies Triage",
      "ignorePaths": ["backend", "infra", "frontend"],
      "matchPackagePatterns": [".*"],
      "automerge": false,
    },
    {
      "groupName": "Backend - Dependencies Triage",
      "matchPaths": ["backend"],
      "matchPackagePatterns": [".*"],
      "automerge": false,
    },
    {
      "groupName": "Frontend - Dependencies Triage",
      "matchPaths": ["frontend"],
      "matchPackagePatterns": [".*"],
      "automerge": false,
    },
    {
      "groupName": "Infra - Dependencies Triage",
      "matchPaths": ["infra"],
      "matchPackagePatterns": [".*"],
      "automerge": false,
    },
    {
      "groupName": "GitHub Actions",
      "depTypeList": ["action"],
      // Automerge is disabled since we don't have automated processes to test our CD pipelines
      "automerge": false,
    },
    {
      "groupName": "Backend - License",
      "matchPaths": ["backend"],
      "matchPackagePatterns": ["^github.com/mitchellh/golicense$"],
      // "automerge": true should not have to specify
    }
  ],
}
```

**Reply by viceice** [reply]
Do you have any issues? We can't test all configs, you should point out if there is something wrong.
Or simply fork your repo and test on fork.

**Reply by SamuelCabralCruz** [reply]
As I mentioned above, settings automerge: true at the first level of the config is not taken into consideration. The only way I get it working is by specifying automerge: true at the package rules level.

**Reply by viceice** [reply]
why do you think it's not working? what does the pr body say, should be something like  this:
![image](https://user-images.githubusercontent.com/1798109/143278201-39b31983-2e8a-4b2d-9b82-7f96c31b626d.png)

**Reply by viceice** [reply]
your's should have a schedule of cause.

**Reply by rarkins** [reply]
`automerge: true` at the top level is a bad idea for most. It means you will also automerge major updates, and if there's both a major and non-major update for the same package then the non-major one will be ignored.

**Reply by SamuelCabralCruz** [reply]
So I guess it is more appropriate to define automerge: true for minor, lockFileMaintenance, pin, digest, and patch at top level.
I will try another approach and will give you feedback afterward.

**Reply by rarkins** [reply]
Yes, I'd define automerge for those, using a package rule

**Reply by SamuelCabralCruz** [reply]
Ok, but it is not exactly what I am trying to achieve. My goal is not to have to define automerge in every package rule but to have both package rules and minor config to be combined. Ex:

"patch": {
  "automerge": true
}


github.com/mitchellh/golicense v1.0.0 -> 1.0.1 would enter package rule Backend - License and since I said at root level that all patch can be automerged, then Automerge would be enabled.

however,

github.com/mitchellh/golicense v1.0.0 -> 2.0.0 would enter package rule Backend - License and since it is a major update, then automerge won't be enabled.

Is this a supported use case?

**Reply by rarkins** [reply]
Just define a package rule with matchUpdateTypes and automerge. It will then apply to all packages

**Reply by SamuelCabralCruz** [reply]
I just tried, but it still do not produce the desired result.
I tried two approaches:
1. define automerge for top level: pin, patch, digest, minor
2. define package rule with matchUpdateTypes: [pin, patch, minor, digest], automerge: true

What I want is that automerge settings is being applied ON TOP of other package rules so that the update of dependencies are still being grouped in logical group, but not to have to redefine automerge for all these package rules.

**Reply by SamuelCabralCruz** [reply]
Anyway I thought to the fact that top level automerge is not a good idea, but I still think that with proper CI and test coverage there is no problem to automerge major updates.
Breaking changes will certainly cause the tests to fail anyway and otherwise it just means that breaking changes impact unused part of the third party contract.
Again, overriding automerge to false with proper rules at the place you consider more risky or not tested enough has to be done with care.
So the initial question remains: why is the top level automerge config not working?

**Reply by rarkins** [reply]
Please create a public reproduction repo
